GetMessages/handler.ts

```diff
@@ -50,42 +65,105 @@ type IGetMessagesHandlerResponse =
  * GetMessages expects a FiscalCode as input and returns the Messages
  * as output or a Validation error.
  *
- * TODO: add full results and paging
  */
 type IGetMessagesHandler = (
-  fiscalCode: FiscalCode
+  fiscalCode: FiscalCode,
+  maybePageSize: O.Option<NonNegativeInteger>,
+  maybeEnrichResultData: O.Option<boolean>,
+  maybeMaximumId: O.Option<NonEmptyString>,
+  maybeMinimumId: O.Option<NonEmptyString>
 ) => Promise<IGetMessagesHandlerResponse>;
 
 /**
  * Handles requests for getting all message for a recipient.
  */
-export function GetMessagesHandler(
-  messageModel: MessageModel
-): IGetMessagesHandler {
-  return async fiscalCode => {
-    return pipe(
-      messageModel.findMessages(fiscalCode),
-      TE.map(flattenAsyncIterator),
-      TE.map(_ => filterAsyncIterator(_, isRight)),
-      TE.map(_ => mapAsyncIterator(_, e => e.right)),
-      TE.map(_ => filterAsyncIterator(_, RetrievedNotPendingMessage.is)),
-      TE.map(_ => mapAsyncIterator(_, retrievedMessageToPublic)),
-      TE.bimap(
-        failure => ResponseErrorQuery(failure.kind, failure),
-        ResponseJsonIterator
-      ),
-      TE.toUnion
-    )();
-  };
-}
+export const GetMessagesHandler = (
+  messageModel: MessageModel,
+  serviceModel: ServiceModel,
+  blobService: BlobService
+): IGetMessagesHandler => async (
+  fiscalCode,
+  maybePageSize,
+  maybeEnrichResultData,
+  maybeMaximumId,
+  maybeMinimumId
+) =>
+  pipe(
+    T.Do,
+    T.bind("pageSize", () =>
+      T.of(O.getOrElse(() => defaultPageSize)(maybePageSize))
+    ),
+    T.bind("shouldEnrichResultData", () =>
+      T.of(O.getOrElse(() => false)(maybeEnrichResultData))
+    ),
+    T.bind("maximumId", () => T.of(O.toUndefined(maybeMaximumId))),
+    T.bind("minimumId", () => T.of(O.toUndefined(maybeMinimumId))),
+    T.map(({ pageSize, shouldEnrichResultData, maximumId, minimumId }) =>
+      pipe(
+        messageModel.findMessages(fiscalCode, pageSize, maximumId, minimumId),
```

Comment:
We have to manage `ResponseErrorQuery` in this case.

Reply:
Done in https://github.com/pagopa/io-functions-app/commit/11001b447ffe2a0d9beeb733b32fd23ec43971f2